archive/zchunk.ksy

```diff
@@ -0,0 +1,188 @@
+            checksum_type == checksum_types::sha256 ? 32 :
+            checksum_type == checksum_types::sha512 ? 64 :
+            checksum_type == checksum_types::sha512_128 ? 16 :
+            0
+  header:
+    seq:
+      - id: preface
+        type: preface
+      - id: len_index
+        type: compressed_integer
+      - id: index
+        type: index
+        size: len_index.value
+      - id: signatures
+        type: signatures
+  preface:
+    seq:
+      - id: checksum
+        size: _root.lead.len_checksum
+      - id: flags
+        type: compressed_integer
+      - id: compression_type
+        type: compressed_integer
+      - id: num_optional_element
+        type: compressed_integer
+        if: has_optional_elements
+      - id: optional_elements
+        type: optional_element
+        repeat: expr
+        repeat-expr: num_optional_element.value
+        if: has_optional_elements
+    instances:
+      has_data_streams:
+        value: flags.value & 0b1 == 0b1
+      has_optional_elements:
+        value: flags.value & 0b10 == 0b10
+  optional_element:
+    seq:
+      - id: element_id
+        type: compressed_integer
+      - id: len_data
+        type: compressed_integer
+      - id: data
+        size: len_data.value
+  index:
+    seq:
+      - id: checksum
+        type: compressed_integer
+      - id: num_chunks
+        type: compressed_integer
+      - id: dict_stream
+        type: compressed_integer
+        if: _parent.preface.has_data_streams
+      - id: dict_checksum
+        size: len_checksum
+      - id: len_dict
+        type: compressed_integer
+      - id: len_uncompressed_dict
+        type: compressed_integer
+      - id: chunks_metadata
+        type: chunk(len_checksum, _parent.preface.has_data_streams)
+        repeat: expr
+        repeat-expr: num_chunks.value - 1
+        doc: the number of chunks includes the header, so -1
+    instances:
+      checksum_type:
+        value: checksum.value
+        enum: checksum_types
+      len_checksum:
+        value: |
+            checksum_type == checksum_types::sha1 ? 20 :
+            checksum_type == checksum_types::sha256 ? 32 :
+            checksum_type == checksum_types::sha512 ? 64 :
+            checksum_type == checksum_types::sha512_128 ? 16 :
+            0
+  chunk:
+    params:
+      - id: len_checksum
+        type: u4
+      - id: has_data_streams
+        type: bool
+    seq:
+      - id: chunk_stream
+        type: compressed_integer
+        if: has_data_streams
+      - id: chunk_checksum
+        size: len_checksum
+      - id: len_chunk
+        type: compressed_integer
+      - id: len_uncompressed_chunk
+        type: compressed_integer
+  signatures:
+    seq:
+      - id: num_signatures
+        type: compressed_integer
+      - id: signatures
+        type: signature
+        repeat: expr
+        repeat-expr: num_signatures.value
+  signature:
+    seq:
+      - id: signature_type
```

Comment:
is it an enum?

Reply:
Yes but the documentation says: " This is an integer containing the type of signature.  Currently there are no
 recognized signature types."